File: js/rna-transcription.js
DnaTranscriber = function() {};

DnaTranscriber.prototype.toRna = function(strand) {
  if (strand.split("").some(x => "CGAT".search(x)==-1)) {
    throw "Invalid input"
  };
  var rnaPairs = {"C":"G", "G":"C", "A":"U", "T":"A"}
  var swaps = strand.split("").map(function(x) {
    return rnaPairs[x]
  });
  return swaps.join("");
};

module.exports = DnaTranscriber;

File: ruby/rna_transcription.rb
class Complement

  def self.of_dna(strand)
    transposed = strand.split("").map{|base| PAIRS[base]}.join
    transposed.length == strand.length ? transposed : ''
  end

  PAIRS = {'C'=>'G', 'G'=>'C', 'T'=>'A', 'A'=>'U'}

end
